### DISCUSSION TITLE
Do not follow redirects by default?

### DISCUSSION THREAD

**Discussion by timf** [danielSanchezQ/warp-reverse-proxy#50] [Ideas]
First, thank you to the creator and contributors of this library. I was in the middle of writing a reverse proxy in warp and, lo and behold, found this and it helped. Thank you.

I like that we can override the inner HTTP client. That solved the issue I was having where the application being proxied issued a redirect to the end-user but the proxy followed the redirect instead of passing the response back. That's an issue because it would mean the end-user sees the content from the redirect but it appears to them under the original URL.

The idea I wanted to present is to make it the default behavior to not follow redirects. It seems like not redirecting behind the scenes would be the behavior most people would want from a reverse proxy. Just something to consider.

Tim

---

If anyone else is having the issue, here is a code snippet to override the inner HTTP client:
```rs
use reqwest::redirect::Policy;
use warp_reverse_proxy::CLIENT as PROXY_CLIENT;

let client = reqwest::Client::builder()
    .redirect(Policy::none())
    .build()
    .expect("client couldn't be built");
PROXY_CLIENT.set(client).expect("client couldn't be set");
```

**Comment by danielSanchezQ** [comment]
Hi @timf !

I am down to make it the default behavior. But, I am not really expert in these matters. I would need some reference to actually check that is indeed something that (for example if other libraries does) should be done by default.

Glad if I can be of help!

Thanks!
